src/lib.rs

```diff
@@ -38,3 +38,19 @@ pub fn sr25519_from_seed(seed: &[u8]) -> Vec<u8> {
     .to_half_ed25519_bytes()
     .to_vec()
 }
+
+#[wasm_bindgen]
+pub fn ed25519_pubkey(secret: &[u8]) -> Vec<u8> {
+  use ed25519_zebra::{SigningKey, VerificationKey};
+  let secret = SigningKey::try_from(&secret[0..32]).expect("Invalid secret");
```

Comment:
@tjjfvi I pretty sure this secret extraction is wrong
In `capi` the we pass a 64 byte key which I think it's a ed25519 expanded secret key

The `ecdsa_pubkey` should have the same issue